src/formatDate.js

```diff
@@ -50,7 +50,26 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separator = fromFormat.pop();
+  const dateParts = date.split(separator);
+  const dateMap = {};
+
+  fromFormat.forEach((format, index) => {
+    dateMap[format] = dateParts[index];
+  });
+
+  if (dateMap['YYYY'] && toFormat.includes('YY')) {
+    dateMap['YY'] = dateMap['YYYY'].slice(-2);
+  } else if (dateMap['YY'] && toFormat.includes('YYYY')) {
+    const year = parseInt(dateMap['YY'], 10);
+
+    dateMap['YYYY'] = year < 30 ? `20${dateMap['YY']}` : `19${dateMap['YY']}`;
```

Comment:
The conversion from 'YY' to 'YYYY' is not handling the case where 'YY' is exactly 30. According to the task description, '30' should be converted to '1930', but the current logic will convert it to '2030'.